data/txHandler.go

```diff
@@ -224,6 +227,40 @@ func (handler *TxHandler) processIncomingTxn(rawmsg network.IncomingMessage) net
 	dec := protocol.NewMsgpDecoderBytes(rawmsg.Data)
 	ntx := 0
 	unverifiedTxGroup := make([]transactions.SignedTxn, 1)
+
+	// if the ElasticRateLimiter doesn't have a reservation for this Peer, attempt to make one
```

Comment:
how about moving this into a separate function? this only depends on rawmsg.Sender